test/unit-test/dawn/IIR/TestIIRSerializer.cpp

```diff
@@ -0,0 +1,312 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Compiler/DiagnosticsEngine.h"
+#include "dawn/Compiler/Options.h"
+#include "dawn/IIR/IIR.h"
+#include "dawn/IIR/IIRSerializer.h"
+#include "dawn/Optimizer/OptimizerContext.h"
+#include <gtest/gtest.h>
+
+using namespace dawn;
+
+namespace {
+#define IIR_EARLY_EXIT(value)                                                                      \
+  if(!value)                                                                                       \
+    return value;
+
+#define IIR_EXPECT_IMPL(iir1, iir2, VALUE)                                                         \
+  do {                                                                                             \
+    EXPECT_##VALUE(compareStencilInstantiations(iir1, iir2));                                      \
+  } while(0);
+
+#define IIR_EXPECT_EQ(iir1, iir2) IIR_EXPECT_IMPL((iir1), (iir2), TRUE)
+#define IIR_EXPECT_NE(iir1, iir2) IIR_EXPECT_IMPL((iir1), (iir2), FALSE)
+
+bool compareIIRs(iir::IIR* lhs, iir::IIR* rhs) {
+  IIR_EARLY_EXIT(lhs->checkTreeConsistency());
+  IIR_EARLY_EXIT(rhs->checkTreeConsistency());
+  // checking the stencils
+  for(int stencils = 0, size = lhs->getChildren().size(); stencils < size; ++stencils) {
+    const auto& lhsStencil = lhs->getChild(stencils);
+    const auto& rhsStencil = rhs->getChild(stencils);
+    IIR_EARLY_EXIT((lhsStencil->getStencilAttributes() == rhsStencil->getStencilAttributes()));
+    IIR_EARLY_EXIT((lhsStencil->getStencilID() == rhsStencil->getStencilID()));
+
+    // checking each of the multistages
+    for(int mssidx = 0, mssSize = lhsStencil->getChildren().size(); mssidx < mssSize; ++mssidx) {
+      const auto& lhsMSS = lhsStencil->getChild(mssidx);
+      const auto& rhsMSS = rhsStencil->getChild(mssidx);
+      IIR_EARLY_EXIT((lhsMSS->getLoopOrder() == rhsMSS->getLoopOrder()));
```

Comment:
are you sure this IIR_EARLY_EXIT is doing what you expected? 

IIR_EARLY_EXIT(false) will return true so that the comparison will not fail